src/Core/src/Handlers/View/ViewHandler.Android.cs

```diff
@@ -85,37 +86,43 @@ public static void MapAnchorY(IViewHandler handler, IView view)
 
 			var accessibilityDelegate = ViewCompat.GetAccessibilityDelegate(handler.NativeView as View) as MauiAccessibilityDelegateCompat;
 
-			if (view.Semantics != null && accessibilityDelegate == null)
-			{
-				if (handler.NativeView is not NativeView nativeView)
-					return;
+			if (handler.NativeView is not NativeView nativeView)
+				return;
 
-				if (nativeView is AndroidX.AppCompat.Widget.SearchView sv)
-					nativeView = sv.FindViewById(Resource.Id.search_button)!;
+			nativeView = nativeView.GetSemanticNativeElement();
 
-				if (!string.IsNullOrWhiteSpace(view.Semantics.Hint) || !string.IsNullOrWhiteSpace(view.Semantics.Description))
+			var desc = view.Semantics?.Description;
+			var hint = view.Semantics?.Hint;
+
+			if (!string.IsNullOrWhiteSpace(hint) ||
+				!string.IsNullOrWhiteSpace(desc) ||
+				!string.IsNullOrWhiteSpace(view.AutomationId))
+			{
+				if (accessibilityDelegate == null)
 				{
-					if (accessibilityDelegate == null)
-					{
-						var currentDelegate = ViewCompat.GetAccessibilityDelegate(nativeView);
-						if (currentDelegate is MauiAccessibilityDelegateCompat)
-							currentDelegate = null;
+					var currentDelegate = ViewCompat.GetAccessibilityDelegate(nativeView);
+					if (currentDelegate is MauiAccessibilityDelegateCompat)
+						currentDelegate = null;
 
-						accessibilityDelegate = new MauiAccessibilityDelegateCompat(currentDelegate)
-						{
-							Handler = handler
-						};
+					accessibilityDelegate = new MauiAccessibilityDelegateCompat(currentDelegate)
+					{
+						Handler = handler
+					};
 
-						ViewCompat.SetAccessibilityDelegate(nativeView, accessibilityDelegate);
-					}
-				}
-				else if (accessibilityDelegate != null)
-				{
-					ViewCompat.SetAccessibilityDelegate(nativeView, null);
+					ViewCompat.SetAccessibilityDelegate(nativeView, accessibilityDelegate);
 				}
 
-				if (accessibilityDelegate != null)
+				// We use MauiAccessibilityDelegateCompat to fix the issue of AutomationId on android breaking accessibility
+				// Because AutomationId gets set on the contentDesc we have to clear that out on the accessibility node
```

Comment:
Could you elaborate a bit more on what's going on here? I get what you're saying, but having trouble understanding how the following code matches. How does setting importantForAccessibility to Yes clear out the node?

Reply:
Whoops! I moved the code that used to be here but didn't move the comment :-) I've moved the comment up to the if statement so it should now make more sense

Reply:
Oo makes sense, thanks!!